agent-bridge-datastore/src/main/java/com/newrelic/agent/bridge/datastore/R2dbcOperation.java

```diff
@@ -30,14 +30,18 @@ public class R2dbcOperation {
     }
 
     public static OperationAndTableName extractFrom(String sql) {
+        String strippedSql = COMMENT_PATTERN.matcher(sql).replaceAll("");
+        String upperCaseSql = strippedSql.toUpperCase(); //upper case for case-insensitive non-regex-checks
         try {
-            String strippedSql = COMMENT_PATTERN.matcher(sql).replaceAll("");
             for (Map.Entry<String, Pattern[]> operation : OPERATION_PATTERNS.entrySet()) {
-                for (Pattern pattern : operation.getValue()) {
-                    Matcher matcher = pattern.matcher(strippedSql);
-                    if(matcher.find()) {
-                        String model = matcher.groupCount() > 0 ? removeBrackets(unquoteDatabaseName(matcher.group(1).trim())) : "unknown";
-                        return new OperationAndTableName(operation.getKey(), VALID_METRIC_NAME_MATCHER.matcher(model).matches() ? model : "ParseError");
+                String opName = operation.getKey();
+                if (upperCaseSql.contains(opName)) { //non-regex check before pattern matching
+                    for (Pattern pattern : operation.getValue()) {
+                        Matcher matcher = pattern.matcher(strippedSql);
+                        if (matcher.find()) {
+                            String model = matcher.groupCount() > 0 ? removeBrackets(unquoteDatabaseName(matcher.group(1).trim())) : "unknown";
+                            return new OperationAndTableName(operation.getKey(), VALID_METRIC_NAME_MATCHER.matcher(model).matches() ? model : "ParseError");
```

Comment:
I think we can prevent calling the matcher if model is "unknown".
Since the idea is to decrease the amount of regexes run, I think it makes sense to refactor this.

Reply:
@meiao I went to update this and realized that line 42

```
String model = matcher.groupCount() > 0 ? removeBrackets(unquoteDatabaseName(matcher.group(1).trim())) : "unknown";
```

is redundant. `matcher.groupCount` counts how many capturing groups are in the pattern, not how many matches were found. Since all the patterns have one capturing group, this always evaluates to 1 and "unknown" is never reached. 

Removed in the next commit.